gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/crud_service/access_point/AccessPointCrudServiceImpl.java

```diff
@@ -70,6 +72,10 @@ public void deleteAccessPoints(final AccessPoint.ReferenceType referenceType, fi
                 AccessPointReferenceType.valueOf(referenceType.name()),
                 referenceId
             );
+
+            //TODO: add to repo with deleted status for synchronization and then handle add/remove on deployer?
```

Comment:
I've initially based the access point synchronizer on fetching the latest from access point repository but currently this only works for creation/update and not delete. I'm thinking either to add status for active/deleted on access points in the repository to handle this on the deployer or basing the synchronization update/delete events like done in DictionarySynchronizer. 

I'm thinking the status is more direct/better. Any thoughts?

Reply:
I added a comment in this direction above. I was wondering how you could manage deleted access points only by using a query on the repository. 
Having a status could be easier but that means we will have deleted access points in the database without any way of removing them, not sure it is something we want.

It is required to cover delete because there is actually no update...this is always to delete and create, or we need to update how access points are managed between cockpit and apim.

Reply:
Thinking through possible options:

1a) On update, add an extra db call to check if already existing and if different to decide whether to do update or create
1b) Add property on command from cockpit saying if new or updated access points to know whether to run create or update

With either of those 2 options I think then delete would only be used for cleanup so could possibly be not synchronized as in those cases the environments are disabled anyway. Or could add status in the database for this case but would only have entries for this when actually deleted vs having entries for every update with delete as before. 

2) Status for deleted, keeping delete and create for update but this would end up with possibly lots of delete status in db

3) Use update/delete events for synchronization like in DictionarySynchronizer 

@guillaumelamirand do you have thoughts of best approach? I'm thinking maybe 1b) with delete status for actual delete case

Reply:
The cockpit command contains the access points for a given env, so on apim side, we remove the existing one and create the new one for the given env. There is nothing you can use to check if it is already existing, basically if it is already existing, there is nothing to do on the db and so nothing to do on the gateway.

What needs to be detected on the gateway is the new access point for an env. Could say that if we detect new access points, old ones are outdated and so we undeployed them on the gateway ?

Reply:
For the gateways, it could be a list of access points. Even so, the cockpit only sends new access points to APIM so I don't think you can update the old one as you don't know it.